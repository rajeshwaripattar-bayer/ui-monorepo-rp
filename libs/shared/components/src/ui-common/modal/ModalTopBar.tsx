import styles from './ModalTopBar.module.scss'
import { IconButton, IconButtonProps } from '@element/react-icon-button'
import { TypoDisplay } from '@element/react-typography'
import { ReactNode } from 'react'
export interface ModalTopBarProps {
  title?: string
  exitIconButtonProps?: IconButtonProps
  trailingIconButtonProps?: IconButtonProps
  trailingContent?: ReactNode
  topBarContent?: ReactNode
}

export function ModalTopBar(props: ModalTopBarProps) {
  const { title, topBarContent, exitIconButtonProps } = props
  let trialingContent = props.trailingContent
  if (!trialingContent && props.trailingIconButtonProps) {
    trialingContent = <IconButton {...props.trailingIconButtonProps} />
  }

  return (
    <div className={styles.container}>
      {topBarContent ? (
        topBarContent
      ) : (
        <>
          <IconButton className={styles['modal_back_button']} icon='close' {...exitIconButtonProps} />
          <TypoDisplay level={6}>{title}</TypoDisplay>
          {trialingContent && <div className={styles.trailing_content}>{trialingContent}</div>}
        </>
      )}
    </div>
  )
}

export default ModalTopBar
