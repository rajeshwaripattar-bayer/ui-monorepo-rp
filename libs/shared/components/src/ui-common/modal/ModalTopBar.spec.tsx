import { render } from '@testing-library/react'

import ModalTopBar from './ModalTopBar'

describe('ModalTopBar', () => {
  it('should render title and back button successfully', () => {
    const { getByRole, getByText } = render(<ModalTopBar title='Modal Title' exitIconButtonProps={{ icon: 'back' }} />)
    expect(getByRole('button', { name: 'back' })).toBeDefined()
    expect(getByText('Modal Title')).toBeDefined()
  })

  it('should render trailing icon when provided', () => {
    const wrapper = render(<ModalTopBar title='Modal Title' trailingIconButtonProps={{ icon: 'search' }} />)
    expect(wrapper.getByRole('button', { name: 'search' })).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  it('should not render trailing icon button when trailingContent is provided', () => {
    const { queryByRole, queryByText } = render(
      <ModalTopBar title='Modal Title' trailingContent=<p>some</p> trailingIconButtonProps={{ icon: 'search' }} />
    )
    expect(queryByRole('button', { name: 'search' })).toBeNull()
    expect(queryByText('Some')).toBeDefined()
  })

  it('should not render title when topBarContent is provided', () => {
    const { getByText, queryByText } = render(<ModalTopBar title='Not Show' topBarContent=<p>Some</p> />)
    expect(getByText('Some')).toBeDefined()
    expect(queryByText('Not Show')).toBeNull()
  })
})
