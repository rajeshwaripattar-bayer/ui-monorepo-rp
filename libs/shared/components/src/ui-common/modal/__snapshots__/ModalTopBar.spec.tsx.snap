// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalTopBar should render trailing icon when provided 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <button
          class="lmnt lmnt-icon-button mdc-icon-button lmnt-icon-button--medium modal_back_button mdc-ripple-upgraded--unbounded"
          tabindex="0"
        >
          <i
            aria-hidden="false"
            class="lmnt lmnt-icon material-icons lmnt-icon--medium"
          >
            close
          </i>
        </button>
        <span
          class="mdc-typography lmnt mdc-typography--headline6"
        >
          Modal Title
        </span>
        <div
          class="trailing_content"
        >
          <button
            class="lmnt lmnt-icon-button mdc-icon-button lmnt-icon-button--medium mdc-ripple-upgraded--unbounded"
            tabindex="0"
          >
            <i
              aria-hidden="false"
              class="lmnt lmnt-icon material-icons lmnt-icon--medium"
            >
              search
            </i>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <button
        class="lmnt lmnt-icon-button mdc-icon-button lmnt-icon-button--medium modal_back_button mdc-ripple-upgraded--unbounded"
        tabindex="0"
      >
        <i
          aria-hidden="false"
          class="lmnt lmnt-icon material-icons lmnt-icon--medium"
        >
          close
        </i>
      </button>
      <span
        class="mdc-typography lmnt mdc-typography--headline6"
      >
        Modal Title
      </span>
      <div
        class="trailing_content"
      >
        <button
          class="lmnt lmnt-icon-button mdc-icon-button lmnt-icon-button--medium mdc-ripple-upgraded--unbounded"
          tabindex="0"
        >
          <i
            aria-hidden="false"
            class="lmnt lmnt-icon material-icons lmnt-icon--medium"
          >
            search
          </i>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
